I620 close figure objects in unit test

chonlei opened the pull request:
Close #620.

MichaelClerx commented on pints/tests/test_plot.py:
```diff
@@ -12,6 +12,7 @@
 import unittest
 import numpy as np
 import matplotlib
+import matplotlib.pyplot as plt
```
This needs to go after `matplotlib.use`

chonlei replied:
OK! Yes, I got flake8 complained so I moved it before... Yes, I can do #noqa and move it later!

chonlei commented:
> `matplotlib.use` has to be called before `pyplot` is imported (might require #noqa to stop flake8 from complaining),but otherwise looks great!

Moved it and added #noqa. If it looks good, feel free to merge it.

MichaelClerx commented:
Thanks! I've gone 1 further and moved the pyplot imports into the test functions (I forgot that that was best practice when I reviewed earlier :man_facepalming: )

chonlei commented:
Ha! I started with that and not sure why I moved it out later. 😂
